src/main/java/edu/wpi/first/pathweaver/extensions/ExtensionManager.java

```diff
@@ -0,0 +1,111 @@
+package edu.wpi.first.pathweaver.extensions;
+
+import edu.wpi.first.pathweaver.DuplicateGameException;
+import edu.wpi.first.pathweaver.Game;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public final class ExtensionManager {
+
+  private static final Logger log = Logger.getLogger(ExtensionManager.class.getName());
+
+  private final String directory = System.getProperty("user.home") + "/PathWeaver/Games";
+  private final ExtensionLoader loader = new ExtensionLoader();
+
+  private final List<Game> games = new ArrayList<>();
+
+  private static final ExtensionManager instance = new ExtensionManager(); // NOPMD constant name
+
+  public ExtensionManager() {
+    new File(directory).mkdirs();
+  }
+
+  public static ExtensionManager getInstance() {
+    return instance;
+  }
+
+  /**
+   * Gets a list of the games in the ~/PathWeaver/Games directory.
+   *
+   * @throws IOException if the games directory could not be read
+   */
+  public List<Game> findGames() throws IOException {
+    List<Game> games = new ArrayList<>();
+
+    Files.list(Paths.get(directory))
+        .filter(Files::isDirectory)
+        .map(this::loadGameFromDir)
+        .flatMap(Optional::stream)
+        .forEach(games::add);
+
+    scanForZips().stream()
+        .map(this::loadGameFromZip)
+        .flatMap(Optional::stream)
+        .forEach(games::add);
+    return games;
+  }
+
+  private Optional<Game> loadGameFromDir(Path dir) {
+    try {
+      return Optional.of(loader.loadFromDir(dir));
+    } catch (IOException e) {
+      log.log(Level.WARNING, "Could not load game from " + dir.toAbsolutePath(), e);
+    } catch (DuplicateGameException e) {
```

Comment:
This is output when editing a project currently. This is because it is loading all the games again if you edit a project's preferences.